tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,664 @@
+        tx_port_ptf_id = setup_info[dest_port_type]['dest_port_ptf_id'][1]
+        peer_ip, peer_mac = get_neighbor_info(duthost, tx_port)
+        peer_ip1 = peer_ip
+
+        add_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip)
+
+        tx_port = setup_info[dest_port_type]['dest_port'][2]
+        tx_port_ptf_id = setup_info[dest_port_type]['dest_port_ptf_id'][2]
+        peer_ip, peer_mac = get_neighbor_info(duthost, tx_port)
+        peer_ip2 = peer_ip
+
+        add_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip)
+ 
+        
+        time.sleep(3)
+
+        partial_ptf_runner(setup_info, setup_mirror_session,self.acl_stage(), self.mirror_type(),
+                           src_port = rx_port_ptf_id,
+                           dst_ports = setup_info[dest_port_type]['dest_port_ptf_id'][0])
+ 
+       
+        partial_ptf_runner(setup_info, setup_mirror_session,self.acl_stage(), self.mirror_type(), expect_receive = False,
+                           src_port = rx_port_ptf_id,
+                           dst_ports =  setup_info[dest_port_type]['dest_port_ptf_id'][1] + ',' + setup_info[dest_port_type]['dest_port_ptf_id'][2])
+
+        remove_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip0)
+        
+        time.sleep(3)
+
+        partial_ptf_runner(setup_info, setup_mirror_session,self.acl_stage(), self.mirror_type(), expect_receive = False,
+                           src_port = rx_port_ptf_id,
+                           dst_ports = setup_info[dest_port_type]['dest_port_ptf_id'][0])
+ 
+        partial_ptf_runner(setup_info, setup_mirror_session,self.acl_stage(), self.mirror_type(),
+                           src_port = rx_port_ptf_id,
+                           dst_ports =  setup_info[dest_port_type]['dest_port_ptf_id'][1] + ',' + setup_info[dest_port_type]['dest_port_ptf_id'][2])
+
+        remove_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip1)
+        remove_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip2)
+
+    def test_everflow_case5(self, duthost, setup_info, setup_mirror_session, dest_port_type, partial_ptf_runner):
+
+        """Test case 5 - Policer enforced DSCP value/mask test"""
+
+        rx_port = setup_info[dest_port_type]['src_port']
+        rx_port_ptf_id =  setup_info[dest_port_type] ['src_port_ptf_id']
+        tx_port = setup_info[dest_port_type]['dest_port'][0]
+        tx_port_ptf_id = setup_info[dest_port_type]['dest_port_ptf_id'][0]
+        peer_ip, peer_mac = get_neighbor_info(duthost, tx_port)
+        add_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip)
+
+        duthost.shell("redis-cli -n 4 hmset 'POLICER|TEST_POLICER' meter_type packets mode sr_tcm\
+                        cir 100 cbs 100 red_packet_action drop")
+
+        duthost.command('config mirror_session add TEST_POLICER_SESSION {} {} {} {} --policer TEST_POLICER'.format(
```

Comment:
You might consider (and this might be better off for a later PR honestly) sticking this in a wrapper method, just for readability. This is probably not super intuitive to someone who hasn't worked on Everflow before, but "setup_mirror_policer" would be a little easier to understand. Totally up to you, since this is a port I'm fine leaving this for a later refactor.

Reply:
(this goes for other parts of the test cases too, probably more reason to hold off on making those kinds of changes now)

Reply:
@daall Yes for now I added comments for this particular sections for understanding.